lib/ast-utils.js

```diff
@@ -20,17 +20,17 @@ module.exports = {
     hasFullTypeInformation(context) {
         var hasFullTypeInformation = (
             context && 
-            this.isTypeScriptParserServices(context.parserServices) &&
-            context.parserServices.hasFullTypeInformation === true
+            this.isTypeScriptParserServices(context.sourceCode.parserServices) &&
```

Comment:
This could throw if context.sourceCode was null - it looks like this file is doing pretty exhaustive checks so may be worth adding that one too